activerecord/lib/active_record/fixtures.rb

```diff
@@ -883,7 +883,11 @@ def set_fixture_class(class_names = {})
       def fixtures(*fixture_set_names)
         if fixture_set_names.first == :all
           fixture_set_names = Dir["#{fixture_path}/{**,*}/*.{yml}"]
-          fixture_set_names.map! { |f| f[(fixture_path.to_s.size + 1)..-5] }
+          fixture_path_globs = Dir[fixture_path].uniq.compact.sort_by!(&:length).reverse!
+          fixture_set_names.map! do |f|
+            glob = fixture_path_globs.find { |glob| f.start_with? glob }
```

Comment:
this might be simpler + more efficient

```
replace = /\A(#{fixture_path_globs.map { |g| Regexp.escape(g) }.join("|")}\/(.+)\.yml\z)/
fixture_set_names.map! { |f| f.sub!(replace, "\\1") }
```